backend/src/cusco/sources/contracts.py

```
@@ -30,45 +32,68 @@ class ContractsSource(DataSource):
             try:
-                data = await self._load_year(year)
-                self._index_contracts(data)
+                contracts = await self._load_year(year)
+                self._extract_index_rows(contracts, supplier_rows, entity_rows)
             except Exception as e:
                 logger.warning(f"Failed to load contracts for {year}: {e}")
 
+        try:
+            await asyncio.to_thread(self._supplier_table.replace_all, supplier_rows)
+            await asyncio.to_thread(self._entity_table.replace_all, entity_rows)
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)